### PR TITLE
Improve yast2_samba.pm

### PR DESCRIPTION
**Pull request by Zaoliang**
- use "dmesg -n 4" against kernel message floating over console
- use wait_screen_change to avoid timeout issue
- see ticket for more information: https://progress.opensuse.org/issues/19922
- see my reference tests:
openSUSE TW:  http://e13.suse.de/tests/3730#step/yast2_samba
openSUSE Leap:  http://e13.suse.de/tests/3731#step/yast2_samba
SLES 12 SP3:  http://e13.suse.de/tests/3750#step/yast2_samba

### CODE REVIEW

**Comment by okurz** [comment]
I only see a dmesg call so the commit and PR description does not fit the content

**Comment by Zaoliang** [comment]
@okurz it seems that the changes got somehow lost, I'm checking this...

**Comment by Zaoliang** [comment]
@okurz got lost changes back :)

File: tests/console/yast2_samba.pm [hunk 1]
```diff
@@ -92,138 +93,125 @@ sub run {
 
     # open port in firewall if it is enabled and check network interfaces, check long text by send key right.
     assert_screen 'yast2_samba_open_port_firewall';
-    send_key 'alt-f';
-    send_key 'alt-i';
+    wait_screen_change { send_key 'alt-f' };
+    wait_screen_change { send_key 'alt-i' };
```
Comment (okurz) [inline tests/console/yast2_samba.pm]: no wait_screen_change before assert_screen

File: tests/console/yast2_samba.pm [hunk 2]
```diff
@@ -92,138 +93,125 @@ sub run {
...
-    wait_still_screen;
     send_key 'alt-t';
     assert_screen 'yast2_samba-server_ldap_test-connection';
-    send_key 'alt-o';
-    wait_still_screen;
+    wait_screen_change { send_key 'alt-o' };
 
     # finally, close with OK
-    send_key 'alt-o';
+    wait_screen_change { send_key 'alt-o' };
```
Comment (okurz) [inline tests/console/yast2_samba.pm]: also this should not be necessary when we have a wait_serial just afterwards

**Comment by rwx788** [comment]
As we see a lot of failures related to the fact that return code is not there after 60 seconds, I suggest that we increase it to some bigger value, e.g. 180.

**Comment by Zaoliang** [comment]
@rwx788 as we talked already, there is no issue related to timeout (wait_serial) at moment. Since a couple of days I have sometimes failures for samba because of yast2 got crashed. I've reported the issue:
https://bugzilla.opensuse.org/show_bug.cgi?id=1051169

**Comment by okurz** [comment]
@rwx788 should we mark the dmesg with a soft fail?

**Comment by rwx788** [comment]
@okurz sounds good to me. But we need to make up our mind about this in general. Adding "quiet" to boot option should do the trick, for example:
console_loglevel
              Only messages with a log level lower than this value will be
              printed to the console.  The default value for this field is
              DEFAULT_CONSOLE_LOGLEVEL (7), but it is set to 4 if the kernel
              command line contains the word "quiet", 10 if the kernel
              command line contains the word "debug", and to 15 in case of a
              kernel fault (the 10 and 15 are just silly, and equivalent to
              8).  The value of console_loglevel can be set (to a value in
              the range 1-8) by a syslog() call with a type of 8.
@Zaoliang here is the bug reference: https://bugzilla.suse.com/show_bug.cgi?id=1011815

**Comment by okurz** [comment]
ok, I added an according comment to https://progress.opensuse.org/issues/19398 . That should be good enough for now

**Pull request merged**